python/nano/src/bigdl/nano/pytorch/strategies/ddp_spawn.py

```diff
@@ -221,6 +227,68 @@ def setup(self, trainer: "pl.Trainer") -> None:
             # so we need to set it to `False` manuallay
             self.model.eval()
 
+        if trainer.training and self.auto_lr:
+
+            def _unpack_lightning_optimizer(opt):
+                return opt._optimizer if isinstance(opt, LightningOptimizer) else opt
+
+            optimizers = self.optimizers
+            optimizers = [_unpack_lightning_optimizer(opt) for opt in optimizers]
+
+            for optimizer in optimizers:
+                for param_group in optimizer.param_groups:
+                    param_group["lr"] *= self.world_size
+
+            lr_scheduler_configs = self.lr_scheduler_configs
+            for config in lr_scheduler_configs:
+                scheduler = config.scheduler
+                scheduler.base_lrs = [lr * self.world_size for lr in scheduler.base_lrs]
+
+            if self.lr_scheduler_configs:
+                warnings.warn(f"Nano warmup currently only support none optimizers, "
+                              f"but got {len(self.lr_scheduler_configs)}. Skip warmup")
+            else:
+                lr_schedulers = []
+                for opt_idx, opt in enumerate(self.optimizers):
+                    if type(self.auto_lr) is bool:
+                        self.auto_lr = trainer.max_epochs // 5 + 1
+                    if not TORCH_VERSION_LESS_1_10:
+                        from torch.optim.lr_scheduler import LinearLR
+                        scheduler = LinearLR(optimizer=opt,
+                                             # set initial lr as user lr
+                                             start_factor=1.0 / self.world_size,
+                                             end_factor=1.0,
+                                             total_iters=self.auto_lr)
```

Comment:
Does `total_iters` mean the number of epochs?

Reply:
Yes